cpp/src/DataStorm/DataElementI.cpp

```diff
@@ -54,6 +52,8 @@ DataElementI::DataElementI(TopicI* parent, const string& name, int64_t id, const
       _config(make_shared<ElementConfig>()),
       _executor(parent->getInstance()->getCallbackExecutor()),
       _listenerCount(0),
+      _forwarder{Ice::uncheckedCast<SessionPrx>(parent->getInstance()->getCollocatedForwarder()->add(
+          [this](Ice::ByteSeq e, const Ice::Current& c) { forward(e, c); }))},
```

Comment:
We capture this, instead of "self" so that we can initialize the forwarder in the constructor. The forwarder is unregister in "destroy" implementation.

Reply:
This code requires a comment. It looks dubious: forward uses _subscribers that is not yet initialized. 

It sounds like we count on `forward` to execute "later" (after construction), but in such a way we know this DataElementI is still alive??

Reply:
added a comment